### ISSUE TITLE
Will freezes shortly after start up with NameError

### ISSUE THREAD

**Issue by wontonst** [skoczen/will#322]
Will works for a while after startup, then freezes with the following exception.

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/multiprocessing/process.py", line 254, in _bootstrap
    self.run()
  File "/usr/local/lib/python3.4/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.4/site-packages/will/backends/io_adapters/base.py", line 71, in __start_event_listeners
    self.handle_outgoing_event(pubsub_event.data)
  File "/usr/local/lib/python3.4/site-packages/will/backends/io_adapters/hipchat.py", line 682, in handle_outgoing_event
    if isinstance(passed_room, basestring):
NameError: name 'basestring' is not defined
```

Takes about a minute before the exception pops up.

```
Mac-mini:~ futbol$ docker exec botty pip freeze
APScheduler==2.1.2
args==0.1.0
beautifulsoup4==4.6.0
bottle==0.12.7
certifi==2017.11.5
chardet==3.0.4
CherryPy==3.6.0
clint==0.3.7
dill==0.2.1
dnspython==1.15.0
fuzzywuzzy==0.15.1
hiredis==0.2.0
html2text==2017.10.4
idna==2.6
Jinja2==2.7.3
Markdown==2.3.1
markdownify==0.4.1
MarkupSafe==0.23
meteor-ejson==1.1.0
parsedatetime==1.1.2
plumbum==1.6.4
pyasn1==0.1.7
pyasn1-modules==0.0.5
pycrypto==2.6.1
pyee==5.0.0
pygerduty==0.28
python-ddp==0.1.5
python-Levenshtein==0.12.0
pytz==2017.2
PyYAML==3.10
redis==2.10.6
regex==2017.9.23
requests==2.18.4
six==1.10.0
slackclient==1.1.0
sleekxmpp==1.3.2
urllib3==1.22
websocket-client==0.44.0
will==2.1.0
will-natural==0.2.1.1
ws4py==0.4.2
```

**Comment by skoczen** [comment]
Thanks for these - those are going to be two different issues & fixes.  Mind splitting one into a separate issue?

Tagging #324 for completeness.

**Comment by wontonst** [comment]
other issue moved to #326

**Comment by skoczen** [comment]
Hey @wontonst thanks for your patience - are you still seeing this?  IF so, what version of python are you on?

**Comment by wontonst** [comment]
@skoczen I still get the basestring error

I am on a docker container running `python:3.6.3-alpine`

**Comment by wontonst** [comment]
Hahaha I did this to myself in my previous contribution. Here's a PR to fix this issue!

https://github.com/skoczen/will/pull/343

**Comment by skoczen** [comment]
Heh, thanks for the fix!  Clearly we've got more platforms running than I can properly test.  Pushing this out shortly!

**Comment by wontonst** [comment]
Thanks!

**Comment by skoczen** [comment]
@wontonst this is now released as 2.1.2!
